server/rooms.ts

```diff
@@ -665,6 +666,17 @@ export abstract class BasicRoom {
 			room => !room.settings.isPrivate || includeSecret
 		);
 	}
+	validateTitle(newTitle: string, newID?: string) {
+		if (!newID) newID = toID(newTitle);
+		// `,` is a delimiter used by a lot of /commands
+		// `|` and `[` are delimiters used by the protocol
+		// `-` has special meaning in roomids
+		if (newTitle.includes(',') || newTitle.includes('|') || newTitle.includes('[') || newTitle.includes('-')) {
+			throw new Chat.ErrorMessage("Room titles can't contain any of: ,|[-");
+		}
+		if (newID.length > 225) throw new Chat.ErrorMessage("The given room title is too long.");
```

Comment:
Rather than hardcoding, I think `const MAX_CHATROOM_ID_LENGTH = 225;` should be defined on top of this file.